.github/workflows/workflows/codeql.yml

```diff
@@ -0,0 +1,46 @@
+name: "Code Scanning - Action"
+
+on:
+  push:
+  schedule:
+    - cron: '0 0 * * 0'
+
+jobs:
+  CodeQL-Build:
+
+    strategy:
+      fail-fast: false
+
+
+    # CodeQL runs on ubuntu-latest, windows-latest, and macos-latest
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      # Override language selection by uncommenting this and choosing your languages
+      # with:
+      #   languages: go, javascript, csharp, python, cpp, java
+
+    # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+    # If this step fails, then you should remove it and run the build manually (see below).
+    - name: Autobuild
+      uses: github/codeql-action/autobuild@v1
```

Comment:
In attempting the LGTM integration, I found that the build step was pretty complex for C++ in our case since we use Bazel to build, and afaik autobuild will support autoconf, make, ninja, and some others.

Reply:
Looks like that's indeed the case. Do you have a build command you've worked out for LGTM?

Reply:
Sent to you over email, but here it is. It gets stuck in our external dependencies build with rules_foreign_cc:
```
0-04-28 13:39:55] [build] ERROR: /opt/work/.cache/bazel/_bazel_semmle-build/b708a88765e665f62450dd3c4416fdee/external/envoy/bazel/foreign_cc/BUILD:210:1: error executing shell command: '/bin/bash -c function cleanup_function() {
....
[2020-04-28 13:39:55] [build] rules_foreign_cc: Printing build logs:
[2020-04-28 13:39:55] [build] _____ BEGIN BUILD LOGS _____
[2020-04-28 13:39:55] [build] bazel-out/host/bin/external/envoy/bazel/foreign_cc/zlib/logs/CMake_script.sh: 1: Syntax error: "(" unexpected
[2020-04-28 13:39:55] [build] _____ END BUILD LOGS ____
```
I confirmed that bash was the default and current shell in the lgtm worker that we were using, and it looks like it attempted to use bash to execute the script. I'm confused why I don't hit that error anywhere else. Maybe @lizan knows?

Here's the config:
```
path_classifiers:
  test: "test"
  docs:
  - "LICENSE"
  - "CODEOWNERS"
  - "NOTICE"
  - "VERSION"
  - "WORKSPACE"
  - "DCO"
  - "docs"
  - "security"
  generated: "generated_api_shadow"
  library: "boost"
  tools:
  - "tools"
  - "ci"
  - "support"
queries:
- exclude: "*"
- include:
    severity: "error"
    tags:
    - "security"
    - "correctness"
extraction:
  cpp:
    prepare:
      packages:
      - "build-essential"
      - "openjdk-8-jdk"
      - "ninja-build"
      - "make"
      - "libncurses5"
      - "unzip"
      - "curl"
      - "virtualenv"
      - "autoconf"
      - "libtool"
      - "wget"
      - "golang"
      - "python"
      - "automake"
    after_prepare:
    - "urlbase=\"https://github.com/bazelbuild/bazel/releases/download\""
    - "version=3.0.0"
    - "filename=\"bazel-${version}-installer-linux-x86_64.sh\""
    - "wget \"${urlbase}/${version}/${filename}\" -O $filename"
    - "chmod +x $filename"
    - "./$filename --prefix=$LGTM_WORKSPACE --skip-uncompress"
    - "wget https://github.com/Kitware/CMake/releases/download/v3.14.5/cmake-3.14.5-Linux-x86_64.sh"
    - "chmod +x cmake-3.14.5-Linux-x86_64.sh"
    - "./cmake-3.14.5-Linux-x86_64.sh --skip-license --prefix=$LGTM_WORKSPACE"
    - "mkdir -p $LGTM_WORKSPACE/clang9"
    - "wget https://releases.llvm.org/9.0.0/clang+llvm-9.0.0-x86_64-linux-gnu-ubuntu-18.04.tar.xz\
      \ -P $LGTM_WORKSPACE/clang9"
    - "cd $LGTM_WORKSPACE/clang9"
    - "tar -xf $LGTM_WORKSPACE/clang9/clang+llvm-9.0.0-x86_64-linux-gnu-ubuntu-18.04.tar.xz\
      \ --strip-components 1"
    - "export PATH=$PATH:$HOME/bin"
    - "export PATH=$LGTM_WORKSPACE/bin:$PATH"
    - "export BAZEL=$LGTM_WORKSPACE/bin/bazel"
    - "export CLANG_TOOLCHAIN=$LGTM_WORKSPACE/clang9"
    - "export CC=$CLANG_TOOLCHAIN/bin/clang"
    - "export CXX=$CLANG_TOOLCHAIN/bin/clang++"
    - "export CLANG_FORMAT=$CLANG_TOOLCHAIN/bin/clang-format"
    index:
      build_command: "$BAZEL --host_jvm_args=-Djavax.net.ssl.trustStore=/etc/ssl/certs/java/cacerts\
        \ --host_jvm_args=-Djavax.net.ssl.trustStorePassword=changeit build  --action_env=PATH=$LGTM_WORKSPACE/bin:$PATH\
        \  --action_env=CC --action_env=CXX -c opt --spawn_strategy=local --verbose_failures\
        \ //source/exe:envoy-static"
```